generated/nise/session.py

```diff
@@ -68,6 +72,15 @@ def __init__(self, vi, library, encoding, freeze_it=False):
     def __repr__(self):
         return '{0}.{1}({2})'.format('nise', self.__class__.__name__, self._param_list)
 
+    def __setattr__(self, key, value):
+        if self._is_frozen and key not in dir(self):
+            raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
+        object.__setattr__(self, key, value)
+
+    def __getitem__(self, key):
+        rep_cap_help_text = ''
+        raise TypeError("'Session' object does not support indexing." + rep_cap_help_text)
```

Comment:
Whats the goal with these error checks?

Reply:
So that they do not try `scope[0].range = 4.0`. If you do that on a `niscope.Session()` there will be a list of repeated capabilities that you might want to use.

`nise` does not have any repeated capabilities (similar to `nidmm`) so the error report would have said to try one of these repeated capability objects and then given an empty list.